tensorizer/stream_io.py

```diff
@@ -417,6 +422,292 @@ def seek(self, position, whence=SEEK_SET):
             self.__init__(self._uri, position, None)
 
 
+def _parse_redis_uri(uri):
+    uri_components = urlparse(uri)
+
+    if uri_components.scheme.lower() != "redis":
+        raise ValueError(f"Invalid Redis URI: {uri}")
+
+    host = uri_components.hostname
+    port = uri_components.port
+    if port is None:
+        port = 6379
+    prefix = uri_components.path.lstrip("/")
+
+    return host, port, prefix
+
+
+# Detect if we're running on OSX, and if so, set max buffer size to 1 MiB.
+if sys.platform == "darwin":
+    _MAX_TCP_BUFFER_SIZE = 2 << 20  # 1 MiB if OSX
+else:
+    _MAX_TCP_BUFFER_SIZE = 2 << 22  # 8 MiB
```

Comment:
The "1 MiB" case was actually 2 MiB. The other change is for readability.
```suggestion
if sys.platform == "darwin":
    _MAX_TCP_BUFFER_SIZE = 1 << 20  # 1 MiB if OSX
else:
    _MAX_TCP_BUFFER_SIZE = 8 << 20  # 8 MiB
```